### PR TITLE
refactor(options-helper): remove util

### PR DESCRIPTION
**Pull request by dawidzarzycki**
OptionsHelper has been removed.
All references related to OptionsHelper in components/files changed.
Some constants that are used more than once has been moved to
theirs config.js files with following pattern:

fixes: #4188 

`<component-name>.config.js`

All affected files are listed below:
- checkbox
- date
- decimal
- form-field
- input-presentation
- input-icon
- label
- numeral-date
- radio-button
- switch
- textarea
- textbox
- accordion
- action-popover-item
- alert
- story-constants
- batch-selection
- button
- button-toggle
- card
- confirm
- detail
- dialog
- flat-table
- form
- help
- icon
- inline-inputs
- link
- loader
- message
- message-content
- multi-action-button
- menu
- pod
- tile
- popover-container
- portrait
- profile
- row
- select-textbox
- show-edit-pod
- sidebar
- split-button
- step-sequence
- table
- tabs-header
- toast
- tooltip
- validation-cion
- vertical-divider
- image
- action-popover
- accordion
- input-definition
- styled-system-interfaces
- options-helper

### Proposed behaviour
Removing OptionsHelper and update all files that uses its reference

### Current behaviour
OptionsHelper exist

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### Additional context
OptionsHelper never showed properly in storybook instead of values it shows OptionsHelper name

### Testing instructions
Suggestions:
- It would be cool to take a look at all affected files in storybook (listed above)
- It will be worth as well to take a look at rest of the components as I could miss something while refactoring so many files

### CODE REVIEW

File: src/__experimental__/components/input/input-presentation.spec.js [hunk 1]
```diff
@@ -31,7 +30,7 @@ describe("InputPresentation", () => {
 
   describe("style", () => {
     describe("sizes", () => {
-      OptionsHelper.sizesRestricted.forEach((size) => {
+      ["small", "medium", "large"].forEach((size) => {
```
Comment (grabkowski) [inline src/__experimental__/components/input/input-presentation.spec.js:33]: that could be replaced with `describe.each`

File: src/__experimental__/components/numeral-date/numeral-date.component.js [hunk 2]
```diff
@@ -241,7 +248,7 @@ NumeralDate.propTypes = {
     const dateFormat = props[propName];
     const isAllowed =
       !dateFormat ||
-      OptionsHelper.dateFormats.find(
+      DATE_FORMATS.find(
```
Comment (grabkowski) [inline src/__experimental__/components/numeral-date/numeral-date.component.js:251]: I think that `allowedDateFormats` would be a more descriptive name

File: src/components/action-popover/action-popover.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import * as React from "react";
-import { MarginSpacingProps } from "../../utils/helpers/options-helper";
+import { MarginSpacingProps } from "style/styled-system-interfaces";
```
Comment (grabkowski) [inline src/components/action-popover/action-popover.d.ts:2]:
```suggestion
import { MarginProps } from "styled-system";
```

File: src/style/styled-system-interfaces.d.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  order?: number;
+}
+
+export interface BackgroundProps {
+  background?: string;
+  backgroundImage?: string;
+  backgroundSize?: string;
+  backgroundPosition?: string;
+  backgroundRepeat?: string;
+}
```
Comment (grabkowski) [inline src/style/styled-system-interfaces.d.ts:67]: this file could be removed, as styled system exports it's own interfaces

File: src/components/vertical-divider/vertical-divider-types.d.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  | 91
+  | 92
+  | 93
+  | 94
+  | 95
+  | 96
+  | 97
+  | 98
+  | 99
+  | 100;
```
Comment (grabkowski) [inline src/components/vertical-divider/vertical-divider-types.d.ts:101]: I think that this is an overkill, I would just set the type as number and then left a comment that the range is 1-100

File: src/__experimental__/components/numeral-date/numeral-date.component.js [hunk 6]
```diff
@@ -17,6 +16,14 @@ const marginPropTypes = filterStyledSystemMarginProps(
   styledSystemPropTypes.space
 );
 
+const DATE_FORMATS = [
```
Comment (grabkowski) [inline src/__experimental__/components/numeral-date/numeral-date.component.js:19]: I think that we should use regular case here, as the convention to mark constants with uppercase was used when there were no `const` keyword in js

File: src/components/button-toggle/button-toggle.config.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line import/prefer-default-export
+export const BUTTON_TOGGLE_ICON_SIZES = ["small", "large"];
```
Comment (grabkowski) [inline src/components/button-toggle/button-toggle.config.js:2]: why do we need a separate file for that?

File: src/components/button/button.type.d.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+
+export type ButtonTypes =
+  | "primary"
+  | "secondary"
+  | "tertiary"
+  | "dashed"
+  | "destructive"
+  | "darkBackground";
```
Comment (grabkowski) [inline src/components/button/button.type.d.ts:8]: this could be placed in the main index file

File: src/components/image/image.d.ts [hunk 9]
```diff
@@ -1,10 +1,7 @@
 import * as React from "react";
+import { BackgroundProps, LayoutProps } from "style/styled-system-interfaces";
 import { StyledComponentProps } from "styled-components";
 import { MarginProps } from "styled-system";
```
Comment (grabkowski) [inline src/components/image/image.d.ts:4]: import { BackgroundProps, LayoutProps, MarginProps } from "styled-system";

File: src/components/message/type-icon/type-icon.spec.js [hunk 10]
```diff
@@ -1,16 +1,17 @@
...
   return TestRenderer.create(<TypeIconStyle {...props} />);
 }
 
+const messages = ["info", "error", "success", "warning"];
+
 describe("TypeIcon", () => {
   describe("when rendered", () => {
     describe("with no additional props", () => {
-      OptionsHelper.messages.forEach((variant) => {
+      messages.forEach((variant) => {
```
Comment (grabkowski) [inline src/components/message/type-icon/type-icon.spec.js:14]: describe.each

**Review by grabkowski — CHANGES_REQUESTED** [review]

File: src/components/button-toggle/button-toggle.config.js (hunk 7 above)
Comment (dawidzarzycki) [reply src/components/button-toggle/button-toggle.config.js:2]: We don't need as it is used only in one file :) Good spot

File: src/components/button/button.type.d.ts (hunk 8 above)
Comment (dawidzarzycki) [reply src/components/button/button.type.d.ts:8]: Yup this is just couple of lines, moved it into `button.d.ts`

File: src/__experimental__/components/numeral-date/numeral-date.component.js (hunk 6 above)
Comment (dawidzarzycki) [reply src/__experimental__/components/numeral-date/numeral-date.component.js:19]: TBH as almost everything is `const` , this is more intuitive to me to keep this in uppercase as it shows me like "this variable keeps hardcoded values"

File: src/components/card/card-column/card-column.spec.js [hunk 11]
```diff
@@ -2,9 +2,9 @@ import React from "react";
 import { shallow } from "enzyme";
 import TestRenderer from "react-test-renderer";
 import CardColumn from "./card-column.component";
-import OptionsHelper from "../../../utils/helpers/options-helper/options-helper";
 import { assertStyleMatch } from "../../../__spec_helper__/test-utils";
 import "jest-styled-components";
```
[the remark this answers is not in the document]
Comment (dawidzarzycki) [reply src/components/card/card-column/card-column.spec.js:6]: 👍  I'll take a look at other files and I'll fix all the places

File: src/__experimental__/components/numeral-date/numeral-date.component.js (hunk 2 above)
Comment (dawidzarzycki) [reply src/__experimental__/components/numeral-date/numeral-date.component.js:253]: I'll rename this to ALLOWED_DATE_FORMATS

**Review by ianforster-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by grabkowski — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by grabkowski — DISMISSED** [review]

**Review by ianforster-sage — DISMISSED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by grabkowski — DISMISSED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**